website_sharing_whole_class_mosaic_plot_translation.Rmd

```diff
@@ -0,0 +1,507 @@
+
+* 长度比面积更容易判断，所以尽量使用相同宽度或高度的矩形。
+
+* 更高更瘦的矩形判断效果更好（因为我们更擅长区分长度而不是面积）。
+
+#### 2.93 矩形之间的间隙
+
+* 无间隙=最高效率
+
+然而，间隙可以帮助提高可读性，所以请尝试不同的组合。
+
+* 可在划分处有间隙
+
+* 可以在层次结构中改变间隙大小。
+
+#### 2.94 颜色
+
+* 对细分小组中的比率有好处
+
+* 显示残余（residual）
+
+* 强调特定群体
+
+### 2.10 其他资源
+
+安东尼-尤文的《用R进行图形数据分析》第七章
+
+本文第三部分（见以下详细内容）
+
+## 3 Part 3
+### 3.1 ggmosaic的基础知识
+
+* 用于分类数据的可视化。
+
+* 可以生成条形图、堆叠条形图、马赛克图和双层图。
+
+* 图是分层构造的，所以变量的排序是非常重要的。
+
+* 集成在ggplot2中的geom里，允许分面(facetting)和分层(layering)。
+
+### 3.2 创建gmosaic
+
+ggmosaic主要是使用ggproto和productplots包创建的。
+
+ggproto能够让你在自己的包中扩展ggplot2。
+
+* 使用了productplots包中的数据处理方法。
+
+* 为了绘制geom，需要计算xmin、xmax、ymin和ymax。
+
+### 3.3 ggplot2的缺陷
+
+ggplot2不能处理总体数量在变化的变量群。
+
+* 目前的解决方案：写成x=product(x1，x2)来同时读入变量x1和x2。
+
+* product函数。
+
+* 是为了wrapper list而设计的函数。
+
+* 允许它通过代码检测。
+
+这些限制也会导致标签的问题，但这些都可以手动修复。
+
+### 3.4 geom_mosaic：设定规则。
+
+规则如下：
+
+* weight：选择一个权重变量
+
+* x：选择要添加到公式中的变量
+
+* 写为x = product(x1, x2, ...)
+
+* fill : 选择一个要填充的变量
+
+* 如果变量没有在x中被调用，它将被添加到公式的首位。
+
+* conds : 选择一个变量作为条件
+
+写为conds = product(cond1, cond2, ...)
+
+然后通过productplots函数发送这些值，以创建所需分布的公式。
+
+```{r}
+#公式：
+weight ~ fill + x | conds
+```
+
+
+#### 3.41 从规则到代码#
+
+如何写成代码
+
+* weight=1
+
+* x = product(Y, X)
+
+* fill=W
+
+* conds = product(Z)
+
+* 这些美学设置了分配的公式。
+
+#### 3.42 1 ~ X
+
+```{r}
+library(ggplot2)
+library(ggmosaic)
+library(gridExtra)
+ ggplot(data = fly) +
+   geom_mosaic(aes(x = product(RudeToRecline), fill=RudeToRecline), na.rm=TRUE) +
+   labs(x="Is it rude recline? ", title='f(RudeToRecline)') 
+```
+
+#### 3.43 1 ~ Y + X
+
+```{r}
+ggplot(data = fly) +
+   geom_mosaic(aes(x = product(DoYouRecline, RudeToRecline), fill=DoYouRecline), na.rm=TRUE) + 
+  labs(x = "Is it rude recline? ", title='f(DoYouRecline | RudeToRecline) f(RudeToRecline)')
+```
+
+#### 3.44 1 ~ X + Y / Z
+
+```{r}
+ggplot(data = fly) +
+   geom_mosaic(aes(x = product(DoYouRecline, RudeToRecline), fill=DoYouRecline, conds=product(Gender)), na.rm=TRUE, divider=mosaic("v")) +  labs(x = "Is it rude recline? ", title='f(DoYouRecline, RudeToRecline| Gender)')
+```
+
+#### 3.45 Alternative to conditioning: facetting
+
+```{r}
+ggplot(data = fly) +
+   geom_mosaic(aes(x = product(DoYouRecline, RudeToRecline), fill=DoYouRecline), na.rm=TRUE) +  labs(x = "Is it rude recline? ", title='f(DoYouRecline, RudeToRecline| Gender)') + facet_grid(Gender~.)
+```
+
+
+### 3.5 Importance of ordering
+
+```{r}
+ ggplot(data = fly) +
+   geom_mosaic(aes(x = product(DoYouRecline, RudeToRecline), fill=DoYouRecline), na.rm=TRUE) +  labs(x = "Is it rude recline? ", title='f(DoYouRecline | RudeToRecline) f(RudeToRecline)') + theme(plot.title = element_text(size = rel(1)))
+
+ ggplot(data = fly) +
+   geom_mosaic(aes(x = product(RudeToRecline, DoYouRecline), fill=DoYouRecline), na.rm=TRUE) + labs(x = "" , y = "Is it rude recline? ", title='f(DoYouRecline | RudeToRecline) f(RudeToRecline)') + coord_flip() + theme(plot.title = element_text(size = rel(1)))
+
+par(mfrow=c(2,1))
+```
+
+
+
+
+### 3.6 geom_mosaic的其他特征
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.